core/model/edition.py

```diff
@@ -724,6 +724,17 @@ def calculate_author(self):
             sort_author = " ; ".join(sorted(sort_names))
         else:
             sort_author = self.UNKNOWN_AUTHOR
+
+        def truncate_string(mystr: str):
+            if len(mystr) > 1024:
```

Comment:
Is there a rationale for choosing 1024 here? Is this the max allowed by a index?

Reply:
I wasn't sure if the 2704 covered just the author (or sort_author) or included other metadata.  So I chose 1024 as a size large enough to include the complete author name for almost all the books, but a safely limited value in case there are other components of the index size calculation that I'm not aware of.